### PR TITLE
#1963 Custom Templates: When switching between tabs, the focus is not active on the search bar

### PR DESCRIPTION
**Pull request by porcelain11**
Create an additional state member in TemplateDialog, which is changing after tab changed, and pass it as a param to input, and then - as a dependency to useEffect, setting focus.

### CODE REVIEW

File: packages/ketcher-react/src/script/ui/component/form/input.jsx [hunk 1]
```diff
@@ -26,12 +26,15 @@ export function GenericInput({
...
   type = 'text',
   isFocused = false,
+  searchFocusBack = false,
   ...props
 }) {
   const inputRef = useRef(null)
   useEffect(() => {
-    if (inputRef.current && isFocused) inputRef.current.focus()
-  }, [inputRef, isFocused])
+    if (inputRef.current && isFocused) {
```
Comment (Nitvex) [inline packages/ketcher-react/src/script/ui/component/form/input.jsx:34]: Can we do the same logic in tabs, not in GenericInput? It belongs to tabs changing, but not for input itself.
I see the solution the following way in tabs component:
```
const inputRef = useRef(null)
...
const onTabChange = () => {
  ... 
  inputRef.current.focus();
}
...
return (
 ...
 <GenericInput ref={inputRef} />
 ...
)
```

Comment (porcelain11) [reply packages/ketcher-react/src/script/ui/component/form/input.jsx:54]: Fixed

**Comment by KonstantinEpam23** [comment]
@porcelain11 hmm, this should be against master, not release branch, the corresponding task is on 2.8.0 milestone

File: packages/ketcher-react/src/script/ui/dialog/template/TemplateDialog.tsx [hunk 2]
```diff
@@ -172,6 +174,10 @@ const TemplateDialog: FC<Props> = (props) => {
     setFilteredFG(filterFGLib(functionalGroups, filter)[FUNCTIONAL_GROUPS])
   }, [functionalGroups, filter])
 
+  useEffect(() => {
+    searchInputRef.current?.focus()
```
Comment (Nitvex) [inline packages/ketcher-react/src/script/ui/dialog/template/TemplateDialog.tsx:178]: May be we can simply move this to `handleTabChange`?

**Review by Nitvex — CHANGES_REQUESTED** [review]

File: packages/ketcher-core/src/application/formatters/identifyStructFormat.ts [hunk 3]
```diff
@@ -77,7 +77,10 @@ export function identifyStructFormat(
     return SupportedFormat.inChI
   }
 
-  if (sanitizedString.indexOf('\n') === -1) {
+  if (
+    sanitizedString.indexOf('\n') === -1 &&
+    sanitizedString === sanitizedString.toUpperCase()
```
Comment (Nitvex) [inline packages/ketcher-core/src/application/formatters/identifyStructFormat.ts:82]: It shouldn't be here. This check was removed

File: packages/ketcher-core/package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "ketcher-core",
-  "version": "2.6.4",
+  "version": "2.7.0-rc.4",
```
Comment (Nitvex) [inline packages/ketcher-core/package.json:3]: Are you sure, we need this? 
In master we still have 2.6.4, in release there is a 2.7.0-rc.8

**Review by Nitvex — CHANGES_REQUESTED** [review]

**Pull request closed without merging**